videostore/utils.go

```diff
@@ -253,31 +254,31 @@ func sortFilesByDate(files []fileWithDate) {
 }
 
 // extractDateTimeFromFilename extracts the date and time from the filename.
+// Returns UTC time regardless of filename format:
+// - Unix timestamp filenames: directly converts to UTC time
+// - Datetime format filenames (YYYY-MM-DD_HH-mm-ss): parses as local time, converts to UTC
 func extractDateTimeFromFilename(filePath string) (time.Time, error) {
-	const minParts = 2
 	baseName := filepath.Base(filePath)
-	parts := strings.Split(baseName, "_")
-	if len(parts) < minParts {
-		return time.Time{}, fmt.Errorf("invalid file name: %s", baseName)
+	nameWithoutExt := strings.TrimSuffix(baseName, filepath.Ext(baseName))
+
+	// Unix timestamp case
+	if timestamp, err := strconv.ParseInt(nameWithoutExt, 10, 64); err == nil {
+		return time.Unix(timestamp, 0).UTC(), nil
```

Comment:
are we sure that the unix timestamps have second precision?

Reply:
I'm quite confident this will return down to the second always. There are other methods for more precision e.g. https://cs.opensource.google/go/go/+/go1.24.1:src/time/time.go;l=1663